### PR TITLE
fix: avoid using closeWallet as a promise

### PR DESCRIPTION
**Pull request by banklesss**

### CODE REVIEW

**Comment by banklesss** [comment]
Last reviews from this [PR](https://github.com/Emurgo/yoroi-mobile/pull/2072) applied here.

@wolverineks

File: src/hooks/index.ts [hunk 1]
```diff
@@ -59,6 +60,26 @@ export const useWallet = (wallet: YoroiWallet, event: WalletEvent['type']) => {
...
+export const useCloseWallet = (options: UseMutationOptions<void, Error> = {}) => {
+  const setSelectedWallet = useSetSelectedWallet()
+  const setSelectedWalletMeta = useSetSelectedWalletMeta()
+
+  const mutation = useMutation({
+    mutationFn: () => walletManager.closeWallet(),
+    onSuccess: (data, variables, context) => {
+      setSelectedWallet(undefined)
+      setSelectedWalletMeta(undefined)
+      options.onSuccess?.(data, variables, context)
```
Comment (wolverineks) [inline src/hooks/index.ts]: destructure callbacks or `...options` will overwrite

Comment (banklesss) [reply src/hooks/index.ts]: done

File: src/Catalyst/Step4.tsx [hunk 2]
```diff
@@ -76,21 +97,15 @@ export const Step4 = ({pin, setVotingRegTxData}: Props) => {
...
+            navigation.goBack()
+          },
           instructions: [strings.bioAuthInstructions],
         })
       } catch (error) {
-        if (error instanceof SystemAuthDisabled) {
-          await closeWallet()
-          await showErrorDialog(errorMessages.enableSystemAuthFirst, intl)
-          navigation.navigate('app-root', {screen: 'wallet-selection'})
+        if (error instanceof Error) {
```
Comment (wolverineks) [inline src/Catalyst/Step4.tsx:106]: i thought we decided to keep this as is?

File: src/Dashboard/WithdrawStakingRewards/ConfirmTx/ConfirmTxWithOS.tsx [hunk 3]
```diff
@@ -51,14 +72,7 @@ export const ConfirmTxWithOS: React.FC<Props> = ({wallet, unsignedTx, onSuccess,
...
               )
-              .catch(async (error) => {
-                if (error instanceof SystemAuthDisabled) {
-                  onCancel()
-                  await closeWallet()
-                  await showErrorDialog(errorMessages.enableSystemAuthFirst, intl)
-                  navigation.navigate('app-root', {screen: 'wallet-selection'})
-                }
-              }),
+              .catch(closeWallet),
```
Comment (wolverineks) [inline src/Dashboard/WithdrawStakingRewards/ConfirmTx/ConfirmTxWithOS.tsx:75]: keep as is?

File: src/SelectedWallet/WalletSelection/WalletSelectionScreen.tsx [hunk 4]
```diff
@@ -45,20 +70,12 @@ export const WalletSelectionScreen = () => {
...
-      if (error instanceof SystemAuthDisabled) {
-        await closeWallet()
-        await showErrorDialog(errorMessages.enableSystemAuthFirst, intl)
-        resetToWalletSelection()
-      } else if (error instanceof InvalidState) {
-        await closeWallet()
-        await showErrorDialog(errorMessages.walletStateInvalid, intl)
-        resetToWalletSelection()
-      } else if (error instanceof KeysAreInvalid) {
+      if (error instanceof KeysAreInvalid) {
```
Comment (wolverineks) [inline src/SelectedWallet/WalletSelection/WalletSelectionScreen.tsx:73]: keep as is?

**Comment by banklesss** [comment]
Closed. Simplified scope here: https://github.com/Emurgo/yoroi-mobile/pull/2131

**Pull request closed without merging**